chore: made adjustments to the mutation to check if the url is valid,…

OlusanyaJegs opened the pull request:
I made adjustments to the mutation to check if the url is valid, append a short 6 character code and also return the original url in a query

OlusanyaJegs committed:
chore: made adjustments to the mutation to check if the url is valid, create a short code and also return the original url

toluaj commented on index.js:
```diff
@@ -46,11 +49,30 @@ const urlMap = {};
     shortenUrl: ({ url }) => {
+
+        if (!validUrl.isUri(url)) {
+            throw new Error("This URL is Invalid.");
```
@OlusanyaJegs Have you tested that this doesn't work with urls that don't exist but have a right format?

OlusanyaJegs replied:
I have tested it but the condition wasn't working as it should. It validates any url that has "https" and invalidates without it.

Same validation issue i was having when i used the "url-exists" npm package

toluaj commented on index.js:
```diff
@@ -22,7 +24,8 @@ const schema = buildSchema(`
     type Mutation {
-        shortenUrl(url: String!): String
+        shortenUrl(url: String!): String,
+        originalUrl(id: String!): String
```
@OlusanyaJegs You can return the originalUrl in the shortenUrl mutation. This means it'll be an object and not a String anymore.

OlusanyaJegs replied:
I tried to do that before and i was getting the error message below

"Error: Unknown type "originalUrl".

OlusanyaJegs committed:
refactored mutations and url validation ammended

OlusanyaJegs commented:
### Description

- Shorten Url
- Check the Url being shortened for validity
- Append a 6 character string to the shortened Url

### Motivation and Context
[Url Shortener](https://www.notion.so/backdrop-photo/Project-1-Url-Shortener-2b6d841d53064c35b2103746f1020397)

### How Has This Been Tested?
It has been tested on the graphql playground

### Screen recording (compulsory)
[Url Shortener](https://www.loom.com/share/a3b14a2ea9cf46f1ac9424d5aa116d55)

### Checklist

- [x] I have created a branch for this patch/feature.
- [ ] My pull request addresses exactly one patch/feature.
- [ ] Each individual commit in the pull request is meaningful.
- [ ] I have added tests to cover my changes.
- [ ] I have created a review app to manually see my changes in action.
- [x] I have added video evidence of my manual tests.
- [x] All new and existing tests passed.
- [x] My code follows the code style of this project.
